src/Misc.pm

```diff
@@ -3034,7 +3034,11 @@ sub processNameRequestQueue {
 			next;
 		}
 
-		next if ($actor->{avoid});
```

Comment:
I was also experiencing this hang. I completely removed the line and have had no issues but showing an error message is probably preferred.